webpack/assets/javascripts/react_app/components/DiffView/DiffView.js

```diff
@@ -14,24 +14,47 @@ const getDiff = (oldText, newText) => {
 };
 
 const DiffView = ({
-  oldText,
-  newText,
-  viewType,
+  oldText, newText, viewType, patch,
 }) => {
   const markEdits = markCharacterEdits({ threshold: 30, markLongDistanceDiff: true });
-  const gitDiff = getDiff(oldText, newText);
-  const files = parseDiff(gitDiff);
-  const hunk = files[0].hunks;
 
-  return (
-    hunk && <Diff hunks={hunk} markEdits={markEdits} viewType={viewType} />
+  // old,new Text
+  if (patch === '') {
+    const gitDiff = getDiff(oldText, newText);
+    const files = parseDiff(gitDiff);
+    const hunk = files[0].hunks;
+
+    return hunk && <Diff hunks={hunk} markEdits={markEdits} viewType={viewType} />;
+  }
+  // Patch
```

Comment:
should we check we actually have a patch?

Reply:
i don't think the component should even be rendered in this case. I think its the responsibility of the parent.